Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,551 @@
+
+instance [AddCommMonoid β] [ContinuousAdd β] {R : Type*} [Semiring R] [Module R β]
+    [ContinuousConstSMul R β] : Module R C_c(α, β) :=
+  Function.Injective.module R ⟨⟨_, coe_zero⟩, coe_add⟩ DFunLike.coe_injective coe_smul
+
+instance [NonUnitalNonAssocSemiring β] [TopologicalSemiring β] :
+    NonUnitalNonAssocSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalNonAssocSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance [NonUnitalSemiring β] [TopologicalSemiring β] :
+    NonUnitalSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance [NonUnitalCommSemiring β] [TopologicalSemiring β] :
+    NonUnitalCommSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance [NonUnitalNonAssocRing β] [TopologicalRing β] :
+    NonUnitalNonAssocRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalNonAssocRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance instNonUnitalRing [NonUnitalRing β] [TopologicalRing β] : NonUnitalRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalRing _ coe_zero coe_add coe_mul coe_neg coe_sub (fun _ _ => rfl)
+    fun _ _ => rfl
+
+instance instNonUnitalCommRing [NonUnitalCommRing β] [TopologicalRing β] :
+    NonUnitalCommRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [IsScalarTower R β β] :
+    IsScalarTower R C_c(α, β) C_c(α, β) where
+  smul_assoc r f g := by
+    ext
+    simp only [smul_eq_mul, coe_mul, coe_smul, Pi.mul_apply, Pi.smul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_assoc]
+
+instance {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [SMulCommClass R β β] :
+    SMulCommClass R C_c(α, β) C_c(α, β) where
+  smul_comm r f g := by
+    ext
+    simp only [smul_eq_mul, coe_smul, coe_mul, Pi.smul_apply, Pi.mul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_comm]
+
+end AlgebraicStructure
+
+section Uniform
+
+variable [UniformSpace β] [UniformSpace γ] [Zero γ]
+variable [FunLike F β γ] [CompactlySupportedContinuousMapClass F β γ]
+
+theorem uniformContinuous (f : F) : UniformContinuous (f : β → γ) :=
+  (map_continuous f).uniformContinuous_of_tendsto_cocompact
+  (HasCompactSupport.is_zero_at_infty (hasCompactSupport f))
+
+end Uniform
+
+section Star
+
+/-! ### Star structure
+
+It is possible to equip `C_c(α, β)` with a pointwise `star` operation whenever there is a continuous
+`star : β → β` for which `star (0 : β) = 0`. We don't have quite this weak a typeclass, but
+`StarAddMonoid` is close enough.
+
+The `StarAddMonoid` and `NormedStarGroup` classes on `C_c(α, β)` are inherited from their
```

Comment:
There is no `NormedStarGroup` any more, right?

Reply:
yes. I remove this and also the comment on C⋆-ring.